src/hotspot/os_cpu/linux_aarch64/os_linux_aarch64.cpp

```diff
@@ -245,7 +245,10 @@ bool PosixSignals::pd_hotspot_signal_handler(int sig, siginfo_t* info,
           tty->print_cr("trap: %s: (SIGILL)", msg);
         }
 
-        return false; // Fatal error
+        va_list detail_args;
+        VMError::report_and_die(INTERNAL_ERROR, msg, detail_msg, detail_args, thread,
+                                pc, info, NULL, NULL, 0, 0);
+        va_end(detail_args);
       }
```

Comment:
I'm not sure it is ok to revert this code. The fix this is part of (for JDK-8255711) was provided explicitly to re-organize the flow of control for handling of fatal errors. Reverting this code appears to undermine the goal of that issue. I would like to get Thomas Stuefe's (@tstuefe) opinion on whether it is appropriate to abort the JVM here vs returning false before accepting this specific change.

Reply:
Sorry for the delay. I'll take a look.

Reply:
Thanks for finding that, and for pinging me. Please revert this part of the change, I opened a separate issue for it (also affects ppc64): https://bugs.openjdk.java.net/browse/JDK-8259539.

Reply:
Thank you Thomas. I have removed the change from my patch. Would you mind taking another look and approving this patch @adinn ?